File: app/src/main/java/it/liceoarzignano/bold/utils/HelpToast.java
package it.liceoarzignano.bold.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.StringRes;
import android.widget.Toast;

public class HelpToast {
    private static final String HELP_TOAST_PREFS = "HelpToastPrefs";
    public static final String KEY_EVENT_LONG_PRESS = "EventsLongPress";
    public static final String KEY_NEWS_LONG_PRESS = "NewsLongPress";

    public HelpToast(Context context, @StringRes int title, String key) {
        SharedPreferences preferences = context.getSharedPreferences(HELP_TOAST_PREFS,
                Context.MODE_PRIVATE);

        if (preferences.getBoolean(key, false)) {
            return;
        }

        preferences.edit().putBoolean(key, true).apply();
        Toast.makeText(context, title, Toast.LENGTH_LONG).show();

    }
}

File: app/src/main/kotlin/it/liceoarzignano/bold/utils/HelpToast.kt
package it.liceoarzignano.bold.utils

import android.content.Context
import android.widget.Toast
import it.liceoarzignano.bold.R

class HelpToast(context: Context, key: String) {

    init {
        val preferences = context.getSharedPreferences(HELP_TOAST_PREFS, Context.MODE_PRIVATE)

        if (!preferences.getBoolean(key, false)) {
            preferences.edit().putBoolean(key, true).apply()
            Toast.makeText(context, R.string.intro_toast_long_press, Toast.LENGTH_LONG).show()
        }
    }

    companion object {
        private val HELP_TOAST_PREFS = "HelpToastPrefs"
        val KEY_MARK_LONG_PRESS = "MarksLongPress"
        val KEY_EVENT_LONG_PRESS = "EventsLongPress"
        val KEY_NEWS_LONG_PRESS = "NewsLongPress"
    }
}
